src/critics/path_align_critic.cpp

```diff
@@ -66,10 +66,10 @@ void PathAlignCritic::score(CriticData & data)
   {
     auto next = xt::view(P3, xt::all(), xt::range(1, _), xt::range(0, 2));
     auto prev = xt::view(P3, xt::all(), xt::range(_, -1), xt::range(0, 2));
-    auto dist = xt::norm_sq(next - prev, {2});
-    trajectories_lengths = xt::sum(dist, {1});
+    auto dist = xt::eval(xt::norm_sq(next - prev, {2}));
+    trajectories_lengths = xt::sum(dist, {1}, xt::evaluation_strategy::immediate);
   }
-  auto accumulated_path_distances = xt::cumsum(P2_P1_norm_sq);
+  auto accumulated_path_distances = xt::eval(xt::cumsum(P2_P1_norm_sq));
```

Comment:
as far as i know cumsum already returns actual tensor, no need to additional eval, but for consistency, probably ok

Reply:
Yeah to be fair, I added it everywhere and noticed that it helped. I could go back and try to find out, but if it doesn't hurt, I see no reason to spend the time